File: frontend/src/lib/menu-product-image-style.ts
import type { CSSProperties } from 'react';

const SQUARE_ASPECT_MIN = 0.88;
const SQUARE_ASPECT_MAX = 1.14;

/** Auto framing for dish photos in square menu cards (portrait/landscape → contain). */
export function menuProductImageStyle(
  naturalWidth: number,
  naturalHeight: number,
): CSSProperties {
  if (!naturalWidth || !naturalHeight) {
    return { objectFit: 'cover', objectPosition: 'center' };
  }

  const aspect = naturalWidth / naturalHeight;
  if (aspect >= SQUARE_ASPECT_MIN && aspect <= SQUARE_ASPECT_MAX) {
    return { objectFit: 'cover', objectPosition: 'center' };
  }

  return { objectFit: 'contain', objectPosition: 'center' };
}

File: apps/customer_mobile/lib/core/utils/menu_product_image_style.dart
import 'package:flutter/material.dart';

const double _squareAspectMin = 0.88;
const double _squareAspectMax = 1.14;

/// Auto framing for dish photos in square menu cards (portrait/landscape → contain).
BoxFit menuProductImageFit(double naturalWidth, double naturalHeight) {
  if (naturalWidth <= 0 || naturalHeight <= 0) {
    return BoxFit.cover;
  }

  final aspect = naturalWidth / naturalHeight;
  if (aspect >= _squareAspectMin && aspect <= _squareAspectMax) {
    return BoxFit.cover;
  }

  return BoxFit.contain;
}
